code/ui/blocks/src/blocks/Source.tsx

```diff
@@ -134,11 +124,7 @@ export const useSourceProps = (
   docsContext: DocsContextProps<any>,
   sourceContext: SourceContextProps
 ): PureSourceProps & SourceStateProps => {
-  const storyIds = props.ids || (props.id ? [props.id] : []);
-  const storiesFromIds = useStories(storyIds, docsContext);
```

Comment:
Seems like this is actually needed. A regression was detected in Chromatic where multiple stories won't result in multiple sources.

With the code that was removed:
<img width="1193" alt="image" src="https://github.com/storybookjs/storybook/assets/1671563/e3dfae79-86fb-4801-8ec9-2056e61effbd">

With the code being removed:
<img width="1184" alt="image" src="https://github.com/storybookjs/storybook/assets/1671563/d11dadf0-c2ee-4491-ba03-7551bbf074f6">


I have no context to know the best path forward here. Please assist @JReinhold .

cc @valentinpalkovic

Reply:
Good catch. I think this needs to stay intact until we remove the deprecated functionality from `Canvas` too. The deprecated Canvas-functionality of showing multiple stories uses this source API under the hood, so they have to be removed together.

Any objections @shilman ?